src/test/java/net/sf/jabref/logic/util/version/VersionTest.java

```diff
@@ -0,0 +1,114 @@
+        assertEquals(BuildInfo.UNKOWN_VERSION, version.getFullVersion());
+    }
+
+    @Test
+    public void versionOneDigit() {
+        String versionText = "1";
+        Version version = new Version(versionText);
+        assertEquals(versionText, version.getFullVersion());
+        assertEquals(1, version.getMajor());
+        assertEquals(0, version.getMinor());
+        assertEquals(0, version.getPatch());
+        assertFalse(version.isDevVersion());
+    }
+
+    @Test
+    public void versionTwoDigits() {
+        String versionText = "1.2";
+        Version version = new Version(versionText);
+        assertEquals(versionText, version.getFullVersion());
+        assertEquals(1, version.getMajor());
+        assertEquals(2, version.getMinor());
+        assertEquals(0, version.getPatch());
+        assertFalse(version.isDevVersion());
+    }
+
+    @Test
+    public void versionThreeDigits() {
+        String versionText = "1.2.3";
+        Version version = new Version(versionText);
+        assertEquals(versionText, version.getFullVersion());
+        assertEquals(1, version.getMajor());
+        assertEquals(2, version.getMinor());
+        assertEquals(3, version.getPatch());
+        assertFalse(version.isDevVersion());
+    }
+
+    @Test
+    public void versionOneDigitDevVersion() {
+        String versionText = "1dev";
+        Version version = new Version(versionText);
+        assertEquals(versionText, version.getFullVersion());
+        assertEquals(1, version.getMajor());
+        assertEquals(0, version.getMinor());
+        assertEquals(0, version.getPatch());
+        assertTrue(version.isDevVersion());
+    }
+
+    @Test
+    public void versionTwoDigitDevVersion() {
+        String versionText = "1.2dev";
+        Version version = new Version(versionText);
+        assertEquals(versionText, version.getFullVersion());
+        assertEquals(1, version.getMajor());
+        assertEquals(2, version.getMinor());
+        assertEquals(0, version.getPatch());
+        assertTrue(version.isDevVersion());
+    }
+
+    @Test
+    public void versionThreeDigitDevVersion() {
+        String versionText = "1.2.3dev";
+        Version version = new Version(versionText);
+        assertEquals(versionText, version.getFullVersion());
+        assertEquals(1, version.getMajor());
+        assertEquals(2, version.getMinor());
+        assertEquals(3, version.getPatch());
+        assertTrue(version.isDevVersion());
+    }
+
+    @Test
+    public void versionNewerThan() {
+        Version olderVersion = new Version("2.4");
+        Version newerVersion = new Version("4.2");
+        assertTrue(newerVersion.isNewerThan(olderVersion));
+    }
+
+    @Test
+    public void versionEqualVersionsWithDev() {
+        Version latestVersion = new Version("4.2");
+        Version currentVersion = new Version("4.3dev");
+        assertFalse(currentVersion.isNewerThan(latestVersion));
```

Comment:
True here?

I would say (please add tests for this):
3.0 dev < 3.0
3.1 dev < 3.1
3.1.1 dev <= 3.1.1
but
3.9 < 4.0 dev
4.0 < 4.1 dev
4.1 < 4.1.1 dev

Or do I miss something?